doc/DESIGN.md

```diff
@@ -283,3 +283,107 @@ const Key  *keyGetMeta(const Key *key, const char* metaName);
 The reason behind this is, that the above functions – as their name
 suggest – only retrieve values. The returned value must not be modified
 directly.
+
+## Design Guidelines Checklist
+
+On potential changes of the API/ABI as detected by
+[`elektra-icheck`](https://build.libelektra.org/job/elektra-icheck),
+please make sure the API has been reviewed according to the
+following 2 checklists:
+
+## Checklist for overall API
+
+### Naming
+
+- [ ] Inconsistent naming of functions or variables
+
+### Consistency
+
+- [ ] Consistent naming schemes
+- [ ] Similar things are named similarly
+- [ ] Different things are named differently
+- [ ] The order of arguments should be consistent across similar functions
+- [ ] Code is formatted according to the [style guidelines](/doc/CODING.md#coding-style)
+
+### Structural Clarity
+
+- [ ] Functions with similar functionality are grouped into the same namespace
+
+### Compatibility
+
+- [ ] All Bindings have been updated to reflect the new API and work properly
+
+### Extensibility
+
+- [ ] New API is easily extensible with additional functionality
+- [ ] Components do not depend too heavily on each other
+
+## Checklist for each function
+
+### Documentation
+
+- [ ] Doxygen Documentation is complete
+      (covers all parameters, brief/short summary, examples)
+- [ ] Change is mentioned in the release notes
+- [ ] Inconsistencies between documentation and code
+- [ ] Inconsistencies between documentation and tests
+- [ ] Proper Documentation of all side effects
+- [ ] Proper Documentation of thread-safety of function
+- [ ] [Symbol versioning](/doc/dev/symbol-versioning.md)
+      is correct for breaking changes
+
+### Naming
+
+- [ ] Abbreviations should be avoided in function / parameter names
+- [ ] Function / parameter names should neither be too long, nor too short
+- [ ] Function name should be clear and unambiguous
+
+### Compatibility
+
+- [ ] ABI/API forward-compatible (breaking backwards-compatibility
+      to add new symbols is fine)
+- [ ] #ifdef present for experimental features
```

Comment:
At the moment we do not have a proper process how to introduce experimental features. The idea of libproposal did not work well. Macros we did not try yet.

Reply:
okay, I only included this because this was part of the deleted API file in the todo folder. Should I just remove and forego this until there is a proper solution in place?

Reply:
Remove it. With this extensive checklist we actually already make quite sure that new additions to the API do not have obvious problems.